### PR TITLE
add apko_tags

### LINKED ISSUE #74: apko_tags data source
Check out this monstrosity:

https://github.com/chainguard-images/images/blob/b39a922351b2c667a6a3ea77bab4762fcc0169f8/main.tf#L69-L73

```
output "package_version" {
  value = [
    for x in module.image.config.contents.packages : regexall("(((([a-z0-9]+)(?:[.][a-z0-9]+)?)(?:[.][a-z0-9]+)?)(?:[-][a-z0-9]+)?)", trimprefix(x, "${var.extract_package}=")) if startswith(x, "${var.extract_package}=")
  ][0][0]
}
```

This takes a resolved image config and a package to extract, and produces prospective tags for the image based on that package's version, like `:1`, `:1.2`, `:1.2.3`, `:1.2.3-r4`.

This should just be a datasource that takes a resolved config and produces a list of strings which can be `for_each`ed over with `oci_tag`.

### PR DESCRIPTION
**Pull request by imjasonh**
Fixes #74 

This copies an internal method from apko, which we should probably move out or export.

### CODE REVIEW

**Comment by imjasonh** [comment]
For future reference, here's how we can filter things out of the results:

```
data "apko_tags" "go" {
  target_package = "go"
}
output "filtered_tags" {
  value = [for each in apko_tags.go.tags : each if each != "1"]
}
```

**Comment by mattmoor** [comment]
@imjasonh I think we can do better than that!
```hcl
data "apko_tags" "go-1.20" {
  target_package = "go-1.20"
}

data "apko_tags" "go-1.19" {
  target_package = "go-1.19"
}

locals {
  go_20 = toset(apko_tags.go-1.20.tags)
  go_19 = toset(apko_tags.go-1.19.tags)
  filtered_go_19 = setsubtract(go_20, go_19)
}

```

File: internal/provider/tags_data_source.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+	popts, ok := req.ProviderData.(*ProviderOpts)
+	if !ok || popts == nil {
+		resp.Diagnostics.AddError("Client Error", "invalid provider data")
+		return
+	}
+	d.popts = *popts
+}
+
+var versionRE = regexp.MustCompile(`^(\d+)\.(\d+)\.(\d+)-r(\d+)$`)
```
Comment (mattmoor) [inline internal/provider/tags_data_source.go:83]: this is unused?

**Review by mattmoor — APPROVED** [review]
Disabled automerge for the dead regex, but otherwise LGTM!

**Pull request merged**